multiversx_sdk_cli/transactions.py

```diff
@@ -323,3 +119,90 @@ def get_guardian_account_from_args(args: Any):
         raise errors.NoWalletProvided()
 
     return account
+
+
+def send_and_wait_for_result(transaction: ITransaction, proxy: INetworkProvider, timeout: int) -> ITransactionOnNetwork:
+    if not transaction.signature:
+        raise errors.TransactionIsNotSigned()
+
+    txOnNetwork = _send_transaction_and_wait_for_result(proxy, transaction, timeout)
+    return txOnNetwork
+
+
+def _send_transaction_and_wait_for_result(proxy: INetworkProvider, payload: ITransaction, num_seconds_timeout: int = 100) -> ITransactionOnNetwork:
+    AWAIT_TRANSACTION_PERIOD = 5
+
+    tx_hash = proxy.send_transaction(payload)
+    num_periods_to_wait = int(num_seconds_timeout / AWAIT_TRANSACTION_PERIOD)
+
+    for _ in range(0, num_periods_to_wait):
+        time.sleep(AWAIT_TRANSACTION_PERIOD)
+
+        tx = proxy.get_transaction(tx_hash)
+        if tx.is_completed:
+            return tx
+        else:
+            logger.info("Transaction not yet done.")
+
+    raise errors.KnownError("Took too long to get transaction.")
+
+
+def tx_to_dictionary_as_inner(tx: Transaction) -> Dict[str, Any]:
+    dictionary = tx.to_dictionary()
+    dictionary["receiver"] = base64.b64encode(bytes.fromhex(tx.receiver.hex())).decode()  # type: ignore
+    dictionary["sender"] = base64.b64encode(bytes.fromhex(tx.sender.hex())).decode()  # type: ignore
+    dictionary["chainID"] = base64.b64encode(tx.chainID.encode()).decode()
+    dictionary["signature"] = base64.b64encode(bytes(bytearray(tx.signature))).decode()
+    dictionary["value"] = tx.value
```

Comment:
what about gas price / gas limit ?

Reply:
`dictionary = tx.to_dictionary()` on the first line of the function. These are the values that need to be overwritten for the relayed transaction.